ndinterp/src/grid.rs

```diff
@@ -53,12 +58,13 @@ impl Grid {
     /// Find the index of the last value in the input such that input(idx) < query
     /// If the query is outside the grid returns an extrapolation error
     pub fn closest_below(&self, query: f64) -> Result<usize, InterpolationError> {
-        if query > self.input[self.input.len() - 1] {
+        if query > self.input[0][self.input[0].len() - 1] {
             Err(InterpolationError::ExtrapolationAbove(query))
-        } else if query < self.input[0] {
+        } else if query < self.input[0][0] {
             Err(InterpolationError::ExtrapolationBelow(query))
         } else {
-            let u_idx = self.input.iter().position(|x| x > &query).unwrap();
+            let u_idx = self.input[0].partition_point(|&x| x < query);
+            //            let u_idx = self.input.iter().position(|x| x > &query).unwrap();
```

Comment:
I'd like to make this function also work for any `d`, but making the output a `Vec<usize>` makes it much slower.

I could instead make the output `[usize; 1]` and that's ok, but I don't know how to make it so that I can do `closest_below<N>(self, query) -> Result<[usize; N]` or something like that. Is that possible in rust?

Reply:
> I could instead make the output `[usize; 1]` and that's ok, but I don't know how to make it so that I can do `closest_below<N>(self, query) -> Result<[usize; N]` or something like that. Is that possible in rust?

That should be possible. Instead of making it generic you probably want to subsitute `N` with the dimension of `Grid`?